src/main/scala/com/raphtory/core/components/querytracker/QueryProgressTracker.scala

```diff
@@ -15,6 +15,65 @@ import org.apache.pulsar.client.api.Schema
 
 import scala.collection.mutable.ListBuffer
 
+
+/**
+  * {s}`QueryProgressTracker`
+  *  : {s}`QueryProgressTracker` is used to track the progress of Raphtory queries in terms of number of perspectives processed and duration taken to process each perspective.
+  *    Queries in Raphtory run as a series of {s}`Perspectives` which are graph views at specific timestamps and windows as the query progresses.
+  *    The progress tracker thus helps track query progress until the job is completed. Query types supported include {s}`PointQuery`, {s}`RangeQuery` and {s}`LiveQuery`
+  *
+  * ## Methods
+  *
+  *   {s}`getJobId(): String`
+  *    : Returns job identifier for the query
+  *
+  *   {s}`getLatestPerspectiveProcessed(): Perspective`
+  *    : Returns the latest {s}`Perspective` processed by the query
+  *
+  *   {s}`getPerspectivesProcessed(): List[Perspective]`
+  *    : Returns list of perspectives processed for the query till the current timestamp
+  *
+  *   {s}`getPerspectiveDurations(): List[Long]`
+  *    : Returns the time duration taken to process each of perspectives till the current timestamp
+  *
+  *   {s}`isJobDone(): Boolean`
+  *    : Checks if job is complete
+  *
+  *   {s}`waitForJob()`
+  *    : Polling mechanism to checks if the job is complete, repeats check after every second
+  *
+  *   {s}`handleMessage(msg: QueryManagement): Unit`
+  *    : Handles message to process the {s}`Perspective` received in case the query is in progress, or {s}`JobDone` if the query is complete
```

Comment:
While not officially private, this should not be part of the public interface, make this a code comment. This probably should be marked `protected`.